sql/src/main/java/org/apache/druid/sql/calcite/schema/NamedSchema.java

```diff
@@ -32,6 +34,12 @@
    */
   String getSchemaName();
 
+  @Nullable
+  default String getSchemaResourceType()
```

Comment:
Would be good to add javadoc here and describe where this is used and the importance of returning non-null if authorization checks are needed on the schema, if my understanding is correct.

Reply:
How about something like `getResourceType(String resource)` as I suggested in my other comment? Even though most implementations will ignore the `resource` parameter for now, it will be useful once we allow different resource types in the same schema.